internal/pkg/consensus/election.go

```diff
@@ -74,37 +88,31 @@ func (em ElectionMachine) GenerateEPoSt(allSectorInfos []abi.SectorInfo, challen
 	return ep.ComputeElectionPoSt(allSectorInfos, challengeSeed, winners)
 }
 
-// VerifyEPoStVrfProof verifies that the PoSt randomness is the result of the
-// candidate signing the ticket.
-func (em ElectionMachine) VerifyEPoStVrfProof(ctx context.Context, base block.TipSetKey, epoch abi.ChainEpoch, miner address.Address, workerSigner address.Address, vrfProof abi.PoStRandomness) error {
+func (em ElectionMachine) VerifyElectionProof(ctx context.Context, entry *drand.Entry, epoch abi.ChainEpoch, miner address.Address, workerSigner address.Address, vrfProof crypto.VRFPi) error {
 	entropy, err := encoding.Encode(miner)
 	if err != nil {
 		return errors.Wrapf(err, "failed to encode entropy")
 	}
-	randomness, err := em.chain.SampleChainRandomness(ctx, base, acrypto.DomainSeparationTag_ElectionPoStChallengeSeed, epoch, entropy)
+	seed := blake2b.Sum256(entry.Signature.Data)
+	randomness, err := crypto.BlendEntropy(acrypto.DomainSeparationTag_ElectionPoStChallengeSeed, seed[:], epoch, entropy)
```

Comment:
If you wrote a vrfRandomness() function that took some bytes (e.g. entry.Signature.Data) an address and chain epoch, you could use it 3 times in this struct (and one more if you ever need to validate a ticket).